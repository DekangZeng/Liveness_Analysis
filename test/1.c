
void test() {
  int a, b, c, d, e, f;
  e = b + c;
  if (e > 0) {
    e = a;
  } else {
    a = b + c;
  }
  a = e + c;
}

int main(int argc, char** argv) {
	test();
  return 0;
}