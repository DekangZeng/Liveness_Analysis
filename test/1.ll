; ModuleID = 'Test.c'
source_filename = "Test.c"
target datalayout = "e-m:e-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-unknown-linux-gnu"

; Function Attrs: noinline nounwind uwtable
define dso_local void @test() #0 {
entry:
  %a = alloca i32, align 4
  %b = alloca i32, align 4
  %c = alloca i32, align 4
  %d = alloca i32, align 4
  %e = alloca i32, align 4
  %f = alloca i32, align 4
  %0 = load i32, i32* %b, align 4
  %1 = load i32, i32* %c, align 4
  %add = add nsw i32 %0, %1
  store i32 %add, i32* %e, align 4
  %2 = load i32, i32* %e, align 4
  %cmp = icmp sgt i32 %2, 0
  br i1 %cmp, label %if.then, label %if.else

if.then:                                          ; preds = %entry
  %3 = load i32, i32* %a, align 4
  store i32 %3, i32* %e, align 4
  br label %if.end

if.else:                                          ; preds = %entry
  %4 = load i32, i32* %b, align 4
  %5 = load i32, i32* %c, align 4
  %add1 = add nsw i32 %4, %5
  store i32 %add1, i32* %a, align 4
  br label %if.end

if.end:                                           ; preds = %if.else, %if.then
  %6 = load i32, i32* %e, align 4
  %7 = load i32, i32* %c, align 4
  %add2 = add nsw i32 %6, %7
  store i32 %add2, i32* %a, align 4
  ret void
}

; Function Attrs: noinline nounwind uwtable
define dso_local i32 @main(i32 %argc, i8** %argv) #0 {
entry:
  %retval = alloca i32, align 4
  %argc.addr = alloca i32, align 4
  %argv.addr = alloca i8**, align 8
  store i32 0, i32* %retval, align 4
  store i32 %argc, i32* %argc.addr, align 4
  store i8** %argv, i8*** %argv.addr, align 8
  call void @test()
  ret i32 0
}

attributes #0 = { noinline nounwind uwtable "correctly-rounded-divide-sqrt-fp-math"="false" "disable-tail-calls"="false" "less-precise-fpmad"="false" "min-legal-vector-width"="0" "no-frame-pointer-elim"="true" "no-frame-pointer-elim-non-leaf" "no-infs-fp-math"="false" "no-jump-tables"="false" "no-nans-fp-math"="false" "no-signed-zeros-fp-math"="false" "no-trapping-math"="false" "stack-protector-buffer-size"="8" "target-cpu"="x86-64" "target-features"="+fxsr,+mmx,+sse,+sse2,+x87" "unsafe-fp-math"="false" "use-soft-float"="false" }

!llvm.module.flags = !{!0}
!llvm.ident = !{!1}

!0 = !{i32 1, !"wchar_size", i32 4}
!1 = !{!"clang version 8.0.1 (tags/RELEASE_801/final)"}
