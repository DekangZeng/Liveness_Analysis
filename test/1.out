----- entry -----
UEVAR: b c 
VARKILL: e 
LIVEOUT: a b c e 
----- if.then -----
UEVAR: a 
VARKILL: e 
LIVEOUT: c e 
----- if.else -----
UEVAR: b c 
VARKILL: a 
LIVEOUT: c e 
----- if.end -----
UEVAR: c e 
VARKILL: a 
LIVEOUT: 